File: src/main/java/dev/xyze/namecolor/util/CommodoreUtil.java
package dev.xyze.namecolor.util;

import com.mojang.brigadier.tree.LiteralCommandNode;
import me.lucko.commodore.Commodore;
import me.lucko.commodore.file.CommodoreFileFormat;
import org.bukkit.plugin.Plugin;

import java.io.IOException;

public class CommodoreUtil {
    public static void registerCommandFromFile(
            final Commodore commodoreInst,
            final Plugin pluginInst,
            final String fileName
    ) throws IOException {
        // this is provided due to kotlin's design of not providing wildcard types
        LiteralCommandNode<?> commandFile = CommodoreFileFormat.parse(pluginInst.getResource(fileName));
        commodoreInst.register(commandFile);
    }
}

File: src/main/kotlin/dev/xyze/namecolor/util/CommodoreUtil.kt
package dev.xyze.namecolor.util

import com.mojang.brigadier.tree.LiteralCommandNode
import me.lucko.commodore.Commodore
import me.lucko.commodore.file.CommodoreFileFormat
import org.bukkit.plugin.Plugin

object CommodoreUtil {
    fun registerCommandFromFile(
            commodoreInst: Commodore,
            pluginInst: Plugin,
            fileName: String
    ) {
        var commandFile = CommodoreFileFormat.parse<Any>(pluginInst.getResource(fileName));
        commodoreInst.register(commandFile);
    }
}